DEVELOPING.md

```diff
@@ -0,0 +1,360 @@
+Purpose
+=======
+
+This document provides useful information for developers and maintainers of SuperCollider. It's a yellow-pages of sorts.
+
+Practices
+=========
+
+Git Workflow
+------------
+
+You can find full information about our git workflow
+[here](https://github.com/supercollider/supercollider/wiki/git-workflow-and-guidelines). In summary, we use the
+_git-flow_ branching model with `master` as the stable branch, `develop` as unstable, and `x.y` branches for release
+branches.
+
+Milestones
+----------
+
+We use four tags to keep track of issues:
+
+- Next patch (example: "3.9.1")
+- Some patch (example: "3.9.x")
+- Next minor (example: "3.10")
+- Some minor (example: "3.x")
+
+Patch-milestoned issues are those that will require bug fixes, while minor-milestoned issues are things like new
+features and major changes that are better left for a minor release. The "Next patch"/"Next minor" milestones mark those
+issues that we've decided _must_ be addressed before the next respective release. They're either the most painful bugs
+or most requested features.
+
+When, in the example milestones above, 3.10 is released, we would move all 3.9.x-milestoned issues to 3.10.x,
+and then collectively decide which issues ought to move from 3.x to 3.11 and from 3.10.x to 3.10.1.
+
+Standards for New UGens
+=======================
+
+UGens proposed for inclusion in the main SuperCollider project should meet more rigorous standards than those found "in
+the wild." These are standards the community has agreed upon for new UGens:
+
+- **General software quality concerns:**
+  - UGens in core should never break backward compatibility.
+  - UGens should be fully documented, with a clear explanation of what it does and an appropriate collection of
+    examples. Be sure to document which UGen inputs are modulatable at which rates.
+  - UGens should be efficient. SuperCollider takes pride in being easy on the CPU, and UGens should help support that
+    reputation.
+  - The UGen should be deemed useful enough to the general SC user base.
+  - Each UGen should be in a separate C++ source file (or multiple source files if it's really that long).
+  - Don't leave any unnecessary print statements lying around.
+- **Safety:**
+  - Check input rates in the sclang class.
+  - UGens should be real-time safe. Ross Bencina's ["Real-time audio programming 101"]
+    (http://www.rossbencina.com/code/real-time-audio-programming-101-time-waits-for-nothing) is required reading.
+    Here is a quick and incomplete blacklist of dangerous features:
+    - `malloc`/`free`/`new`: use `RTAlloc` instead.
+    - `throw`/`catch`: use return codes instead.
+    - `dynamic_cast`
+    - System calls. These are complicated to handle and require use of an NRT thread.
+  - Any calls to `RTAlloc` should be protected from `RTAlloc` returning a null pointer. This usually happens when there
```

Comment:
Better written:
`Any calls to `RTAlloc` should be protected from returning a null pointer.`